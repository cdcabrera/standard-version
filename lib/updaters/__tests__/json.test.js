const { readVersion, writeVersion, isPrivate } = require('../types/json');

describe('json updaters', () => {
  it('should read version information', () => {
    const contents = {
      version: 'lorem ipsum'
    };

    expect(readVersion(JSON.stringify(contents))).toMatchSnapshot('read');
  });

  it('should return version information to write', () => {
    const contents = {
      version: 'lorem ipsum'
    };

    expect(writeVersion(JSON.stringify(contents), 'dolor sit')).toMatchSnapshot('write');
  });

  it('should return the private prop', () => {
    const contents = {
      version: 'lorem ipsum',
      private: 'hello world'
    };

    expect(isPrivate(JSON.stringify(contents))).toMatchSnapshot('isPrivate');
  });
});
