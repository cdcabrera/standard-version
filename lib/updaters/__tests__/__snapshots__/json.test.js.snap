// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`json updaters should read version information: read 1`] = `"lorem ipsum"`;

exports[`json updaters should return the private prop: isPrivate 1`] = `"hello world"`;

exports[`json updaters should return version information to write: write 1`] = `
"{
  \\"version\\": \\"dolor sit\\"
}
"
`;
